src/assignment1.js

```diff
@@ -1,5 +1,115 @@
-window.onload = function() {
-  const canvas = document.getElementById('mainDrawingCanvas');
-  const context = canvas.getContext('2d');
+const validShapes = ['triangle', 'circle', 'rect'];
+const sin60 = Math.sqrt(3) / 2;
 
+window.onload = function () {
+    const canvas = document.getElementById('mainDrawingCanvas');
+    const context = canvas.getContext('2d');
+    const width = canvas.clientWidth;
+    const height = canvas.clientHeight;
+    const figures = ('shape:circle/center:120,310/radius:100/velocity:1,2/color:#64b1' +
+        '|shape:rect/center:256,128/width:78/height:154/velocity:2,3/color:#3381' +
+        '|shape:triangle/center:340,389/length:180/velocity:7,1/color:#19a1' +
+        '|shape:circle/center:500,500/radius:70/velocity:-4,-6/color:#b1b1' +
+        '|shape:rect/center:200,128/width:100/height:84/velocity:2,-4/color:#3441' +
+        '|shape:triangle/center:340,389/length:150/velocity:7,10/color:#49f1')
+        .split(' ').join('').split('|').map((value) => parseFigure(value))
+        .filter((value) => value && value.containsX(width) && value.containsY(height));
+    window.setInterval(function () {
+        context.clearRect(0, 0, width, height);
+        figures.forEach((figure) => {
+            figure.draw(context);
+            figure.center[0] += figure.velocity[0];
+            figure.center[1] += figure.velocity[1];
+            figure.checkIfContainsAndChangeDirection(width, height);
+        });
+    }, 50 / 3);
 };
+
+function parseFigure(string) {
```

Comment:
Try to avoid names like string, String, other primitive or common names for parameters and variables.
In javascript you can even have a variables named `undefined` but that's a really bad idea.
I think it's better to use `source` or `rawString` or something like that.

Reply:
done